squire_core/src/state.rs

```diff
@@ -25,17 +30,46 @@ impl AppState {
 
         let client = DbClient::with_options(client_options).unwrap();
 
-        Self { client }
+        let slf = Self { client };
+
+        let index = IndexModel::builder().keys(doc! {"tourn.id": 1}).build();
+        slf.get_tourns().create_index(index, None).await;
+
+        slf
     }
 
+    #[cfg(not(test))]
     pub fn get_db(&self) -> Database {
         self.client.database("Squire")
     }
 
+    #[cfg(test)]
+    pub fn get_db(&self) -> Database {
```

Comment:
This is a very good idea. I think we will have testing Docker containers, but it is always best to have a second layer of protection.

Reply:
The more that I think about this, we probably want to make these static values configurable via a builder. We should have an `AppState` builder where we can specify things like the MongoDB URL, the main database, and the main tournaments table. Then, if not specified, default to these static values. I imagine that this will be the most helpful during testing since every test will be doing arbitrary operations on a table (including dropping) and we can not (easily) ensure that the tests will run in a particular order. This is something we are running into at work, and I would prefer to not repeat such a headache.